src/capi/impl/capi_visualization.cpp

```diff
@@ -0,0 +1,55 @@
+/* Copyright © 2018 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+#include <capi/TuriCreate.h>
+#include <capi/impl/capi_wrapper_structs.hpp>
+#include <capi/impl/capi_error_handling.hpp>
+#include <capi/impl/capi_initialization_internal.hpp>
+#include <flexible_type/flexible_type.hpp> 
+#include <export.hpp>
+
+extern "C" {
+
+    tc_plot* tc_plot_create_1d(const tc_sarray* sa,
```

Comment:
To future-proof this, I would also pass in a tc_parameters* options to allow you to pack it other options later on as needed.

Reply:
Not sure I am understanding the purpose of this:

* Are C API interfaces within a major release not expected/allowed to change arbitrarily? If not, I think I'd prefer to do some more up-front API design on the whole of visualization C API before proceeding.
* Should all parameters go through a single `tc_parameters*` parameter, or be passed explicitly, or a mix of both? If it's a mix of both, when should a parameter be passed explicitly vs. included in the tc_parameters struct?

Reply:
It's a mix of both -- the main parameters I like to put in explicitly, the ones not really used often I delegate to the optional parameters. 

And yes, once an API is in an internal release, then it has to always be valid. We had some symbol naming tricks to redirect things, so it's not totally set in stone, but it's a pain to deprecate old things.

Reply:
Got it, thanks. I'll add `tc_parameters *` to all new functions in this PR to future-proof, and consider offloading some less-used parameters there.